src/main/java/duke/helper/Parser.java

```diff
@@ -0,0 +1,82 @@
+package duke.helper;
+
+import duke.exception.EmptyTaskException;
+import duke.exception.InvalidCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+public class Parser {
+    /**
+     * Validates the task input, checks for empty tasks. If task is empty, throws EmptyTaskException.
+     *
+     * @param input the user input.
+     * @throws EmptyTaskException
+     */
+    public void validateTaskInput(String input) throws EmptyTaskException {
+        String s = input.strip();
+        if (s.equals("todo") || s.equals("deadline") || s.equals("event")) {
+            throw new EmptyTaskException("Task " + s + " cannot be empty nya!\n");
+        }
+    }
+
+    /**
+     * Creates an appropriate task from the user input. Tasks created can be ToDo, Deadline or Event.
+     *
+     * @param input the user input.
+     * @return Task the resultant task formed.
+     */
+    public Task makeTaskFromInput(String input) {
+        validateTaskInput(input);
+        String rest = input.substring(input.indexOf(" ") + 1);
+        switch (input.split(" ")[0]) {
```

Comment:
Great! Your `switch` and `case` statements are correctly-indented, avoiding the common mistake of indenting the `case` statements.